src/sql-parser/src/parser.rs

```diff
@@ -3883,47 +3883,71 @@ impl<'a> Parser<'a> {
         } else if let Some(object_type) = self.parse_one_of_keywords(&[
             OBJECTS, ROLES, CLUSTERS, SCHEMAS, SINKS, SOURCES, TABLES, TYPES, USERS, VIEWS, SECRETS,
         ]) {
+            let object_type = match object_type {
+                OBJECTS => ObjectType::Object,
+                ROLES | USERS => ObjectType::Role,
+                CLUSTERS => ObjectType::Cluster,
+                SCHEMAS => ObjectType::Schema,
+                SINKS => ObjectType::Sink,
+                SOURCES => ObjectType::Source,
+                TABLES => ObjectType::Table,
+                TYPES => ObjectType::Type,
+                VIEWS => ObjectType::View,
+                SECRETS => ObjectType::Secret,
+                _ => unreachable!(),
+            };
+
+            let (from, in_cluster) = match self.parse_one_of_keywords(&[FROM, IN]) {
+                Some(kw) => {
+                    if matches!(object_type, ObjectType::Sink)
+                        && kw == IN
+                        && self.peek_keyword(CLUSTER)
+                    {
+                        // put `IN` back
+                        self.prev_token();
+                        (None, self.parse_optional_in_cluster()?)
+                    } else {
+                        let from = self.parse_object_name()?;
+                        let in_cluster = self.parse_optional_in_cluster()?;
+                        (Some(from), in_cluster)
+                    }
+                }
+                None => (None, None),
+            };
```

Comment:
I think I'd actually feel better if we make `CLUSTER` a reserved keyword across all `SHOW` statements, so that there isn't a disparity between `SHOW TABLES IN CLUSTER` vs `SHOW SINKS IN CLUSTER`. Specifically, I think `SHOW TABLES IN CLUSTER` should be a syntax error of the form "expected object name, found 'cluster'". Thoughts?